frontend/test/pytest/test_template.py

```diff
@@ -689,8 +997,274 @@ def qnn(weights, inputs):
+        return qml.state()
+
+    device = qml.device(backend, wires=4)
+    interpreted_fn = qml.QNode(select, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_fn(), jitted_fn())
+
+
+def test_controlled_sequence(backend):
+    """Test controlled sequence."""
+
+    def controlled_sequence(x):
+        """Test circuit"""
+        qml.PauliX(2)
+        qml.ControlledSequence(qml.RX(x, wires=3), control=[0, 1, 2])
+        return qml.probs(wires=range(4))
+
+    x = jnp.array(0.25)
+    device = qml.device(backend, wires=4)
+    interpreted_fn = qml.QNode(controlled_sequence, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    assert np.allclose(interpreted_fn(x), jitted_fn(x))
+
+
+def test_fable(backend):
+    """Test FABLE."""
+
+    def fable(input_matrix):
+        qml.FABLE(input_matrix, wires=range(5), tol=0)
+        return qml.expval(qml.PauliZ(wires=0))
+
+    input_matrix = np.array(
+        [
+            [-0.5, -0.4, 0.6, 0.7],
+            [0.9, 0.9, 0.8, 0.9],
+            [0.8, 0.7, 0.9, 0.8],
+            [0.9, 0.7, 0.8, 0.3],
+        ]
+    )
+
+    device = qml.device(backend, wires=5)
+    interpreted_fn = qml.QNode(fable, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    assert np.allclose(interpreted_fn(input_matrix), jitted_fn(input_matrix))
+
+
+def test_qubitization(backend):
+    """Test Qubitization."""
+
+    def qubitization(coeffs):
+        H = qml.ops.LinearCombination(coeffs, [qml.Z(0), qml.Z(1), qml.Z(0) @ qml.Z(2)])
+        qml.Hadamard(wires=0)
+        qml.Qubitization(H, control=[3, 4])
+        return qml.expval(qml.PauliZ(0) @ qml.PauliZ(4))
+
+    coeffs = [0.1, 0.3, -0.3]
+    device = qml.device(backend, wires=5)
+    interpreted_fn = qml.QNode(qubitization, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    assert np.allclose(interpreted_fn(coeffs), jitted_fn(coeffs))
+
+
+def test_qrom(backend):
+    """Test QROM."""
+
+    def qrom():
+        qml.QROM(["1", "0", "0", "1"], control_wires=[0, 1], target_wires=[2], work_wires=[3])
+        return qml.probs(wires=3)
+
+    device = qml.device(backend, wires=4)
+    interpreted_fn = qml.QNode(qrom, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    assert np.allclose(interpreted_fn(), jitted_fn())
+
+
+def test_prepselprep(backend):
+    """Test PrepSelPrep"""
+
+    params = np.array([0.4, 0.5, 0.1, 0.3])
+
+    def prepselprep(coeffs):
+        H = qml.ops.LinearCombination(
+            coeffs, [qml.Y(0), qml.Y(1) @ qml.Y(2), qml.X(0), qml.X(1) @ qml.X(2)]
+        )
+        qml.PrepSelPrep(H, control=(3, 4))
+        return qml.expval(qml.PauliZ(3) @ qml.PauliZ(4))
+
+    device = qml.device(backend, wires=5)
+    interpreted_fn = qml.QNode(prepselprep, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    assert np.allclose(interpreted_fn(params), jitted_fn(params))
+
+
+def test_mod_exp(backend):
```

Comment:
CI seems to say this one is failing?

Reply:
Ahh, for stochastic readouts (like `sample` with finite shots) the test can stochastically fail. We have a seed argument you can provide to qjit. See https://github.com/PennyLaneAI/catalyst/blob/main/frontend/test/pytest/test_seeded_qjit.py#L114 for an example.

Reply:
This example even if we are doing sample, it does not give a storychastic result. The test could be changed to probs or state with no problem. Would it still fail in that case?

Reply:
The test still fails with probs or state. I'm just going to leave this one as xfail.

Reply:
Okey, but please let the catalyst team know as it is a template we want to make qjit compatible since a demo is being done on it